src/main/PDF/PdfController.java

```diff
@@ -69,20 +68,35 @@ public PdfController(MongoDatabase db) {
         ctx.json(res.toString());
       };
 
+  /*
+  REQUIRES JSON Body:
+    - "pdfType": String giving PDF Type ("FORM", "APPLICATION", "IDENTIFICATION")
+    - "fileId": String giving id of file to be downloaded
+  */
   public Handler pdfDownload =
       ctx -> {
-        String user = ctx.sessionAttribute("username");
-        String organizationName = ctx.sessionAttribute("orgName");
-        UserType privilegeLevel = ctx.sessionAttribute("privilegeLevel");
+        String user = Objects.requireNonNull(ctx.sessionAttribute("username"));
+        String organizationName = Objects.requireNonNull(ctx.sessionAttribute("orgName"));
+        UserType privilegeLevel = Objects.requireNonNull(ctx.sessionAttribute("privilegeLevel"));
+
         JSONObject req = new JSONObject(ctx.body());
-        String fileIDStr = req.getString("fileID");
-        PDFType pdfType = PDFType.createFromString(req.getString("pdfType"));
+        String fileIDStr = Objects.requireNonNull(req.getString("fileId"));
+        PDFType pdfType =
+            Objects.requireNonNull(PDFType.createFromString(req.getString("pdfType")));
         ObjectId fileID = new ObjectId(fileIDStr);
+
         if (pdfType == null) {
           ctx.result("Invalid PDFType");
         } else {
           InputStream stream =
-              PdfMongo.download(user, organizationName, privilegeLevel, fileID, pdfType, db);
+              PdfMongo.download(
+                  user,
+                  organizationName,
+                  privilegeLevel,
+                  fileID,
+                  pdfType,
+                  db,
+                  encryptionController);
```

Comment:
EncrpytionController calls decrpyFile() from this method. DecryptFile() can be static. This class does not need an EncryptionController member field, and neither does this method.